auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/service/AltusIAMService.java

```diff
@@ -15,33 +17,67 @@
         this.umsClient = umsClient;
+        this.useSharedAltusCredential = altusDatabusConfiguration.isUseSharedAltusCredential();
+        this.accessKey = altusDatabusConfiguration.getSharedAccessKey();
+        this.secretKey = altusDatabusConfiguration.getSharedSecretKey();
     }
 
     /**
      * Generate machine user with access keys
      */
-    public Optional<AltusCredential> generateMachineUserWithAccessKey(String machineUserName, String actorCrn) {
+    public Optional<AltusCredential> generateMachineUserWithAccessKey(String machineUserName, String actorCrn, boolean useSharedCredential) {
+        if (useSharedAltusCredential) {
+            LOGGER.debug("Use shared altus credential is turned on for generating altus credential and access keys.");
+            if (areDatabusCredentialsFilled(useSharedCredential)) {
+                LOGGER.debug("Access and secret keys are set manually application wide for Databus, skip machine user and access key generation");
+                return Optional.of(new AltusCredential(accessKey, secretKey));
+            }
+        }
+
         return Optional.of(umsClient.createMachineUserAndGenerateKeys(
                 machineUserName,
                 actorCrn,
                 umsClient.getBuiltInDatabusRoleCrn(),
                 UserManagementProto.AccessKeyType.Value.ED25519));
     }
 
+    public Optional<AltusCredential> generateMachineUserWithAccessKey(String machineUserName, String actorCrn) {
+        return generateMachineUserWithAccessKey(machineUserName, actorCrn, false);
+    }
+
     /**
      * Delete machine user with its access keys (and unassign databus role if required)
      */
-    public void clearMachineUser(String machineUserName, String actorCrn) {
+    public void clearMachineUser(String machineUserName, String actorCrn, boolean useSharedCredential) {
         try {
-            umsClient.clearMachineUserWithAccessKeysAndRole(machineUserName, actorCrn, umsClient.getBuiltInDatabusRoleCrn());
+            if (useSharedAltusCredential && areDatabusCredentialsFilled(useSharedCredential)) {
+                LOGGER.debug("Access and secret keys are set manually application wide for Databus, skip machine user cleanup.");
+            } else {
+                umsClient.clearMachineUserWithAccessKeysAndRole(machineUserName, actorCrn, umsClient.getBuiltInDatabusRoleCrn());
+            }
         } catch (Exception e) {
             LOGGER.warn("Cluster Databus resource cleanup failed (fluent - databus user). It is not a fatal issue, "
                     + "but note that you could have remaining UMS resources for your account", e);
         }
     }
 
+    public void clearMachineUser(String machineUserName, String actorCrn) {
+        clearMachineUser(machineUserName, actorCrn, false);
+    }
+
+    private boolean areDatabusCredentialsFilled(boolean useSharedCredential) {
+        return secretKey != null && StringUtils.isNoneBlank(accessKey, new String(secretKey));
```

Comment:
This will execute `new String(secretKey)` every time this method is called, creating a new object unnecessarily. Why not change the type of `secretKey` (the field in this class) to `String` and evaluate `new String(secretKey)` only once in the constructor?